### ISSUE TITLE
Num Lock breaks the spacebar in neovim using enable_kitty_keyboard

### ISSUE THREAD

**Issue by AThePeanut4** [wez/wezterm#3526]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Hyprland

### WezTerm version

wezterm 20230413-142547-36d9e13c

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When enable_kitty_keyboard is used, the space character no longer works when Num Lock or Caps Lock is enabled in Neovim. This is due to the incorrect handling of the "Disambiguate escape codes" enhancement, which is the only enhancement that Neovim requests.

### To Reproduce

I slightly modified the program in #3473 to test it, the only difference being requesting mode 1 and not 15:
```c
#include <termios.h> 
#include <stdio.h> 
#include <unistd.h> 

int main()
{
    struct termios _ts = {};
    int _ts_r = tcgetattr(1, &_ts);
    if (_ts_r == 0) {
        struct termios ts_ne = _ts;
        //ts_ne.c_lflag &= ~(ECHO | ECHONL);
        cfmakeraw(&ts_ne);
        if (tcsetattr(1, TCSADRAIN, &ts_ne ) != 0) {
            perror("TTYBackend: tcsetattr");
        }
    }


    fprintf(stderr, "\x1b[?1049h");
    fprintf(stderr, "\x1b[?1000h");
    fprintf(stderr, "\x1b[?1001h");
    fprintf(stderr, "\x1b[?1002h");

    fprintf(stderr, "\x1b[=1u");

    for (;;) {
        char ch = 0;
        if (read(0, &ch, 1) <= 0) break;
        if (ch <= 0x20 || ch == 0x7f) {
            if (ch == 0x1b) fprintf(stderr, " ");
            fprintf(stderr, "\\x%02x", (unsigned int)(unsigned char)ch);
            } else {
            fprintf(stderr, "%c", ch);
        }
        if (ch == 0x0d) break;
        if (ch == 0x0a) break;
    }
    fprintf(stderr, "\x1b[?1049l");
    fprintf(stderr, "\x1b[?1000l");
    fprintf(stderr, "\x1b[?1001l");
    fprintf(stderr, "\x1b[?1002l");

    fprintf(stderr, "\x1b[=0u");

    if (tcsetattr(1, TCSADRAIN, &_ts ) != 0) {
        perror("TTYBackend: tcsetattr");
    }
}
```

Typing the sequence "\<Num Lock on\>\<space\>\<Num Lock off\>\<space\>" gives the following output
Kitty: \x20 \x20
Wezterm: \x1b[57360;1u \x1b[32;129u \x1b[57360;129u \x20

Wezterm is sending the escape codes for the numlock presses, and encoding the first \<space\> to include the Num Lock modifier instead of just sending a space character. From the keyboard protocol docs [here](https://sw.kovidgoyal.net/kitty/keyboard-protocol/#report-all-keys-as-escape-codes), only mode 8 ("Report all keys as escape codes") should send events for pressing modifier keys themselves, and text-generating keys (i.e. space) should be encoded using the "legacy" encoding if it is not specified.

### Configuration

{
    enable_kitty_keyboard = true
}

**Comment by wez** [comment]
I couldn't reproduce this on 36d9e13c which is the version you mentioned above. Were you running an older version?

**Comment by wez** [comment]
Sorry, had the keyboard that has those keys plugged into the wrong computer; yes, I can verify that behavior on that revision.
Has been resolved now in main!

**Comment by AThePeanut4** [comment]
Thanks for the quick fix, I can confirm that the spacebar now works properly on c544b0edce9b54216c752e4679304795b4b8021a. 

The terminal is still sending through the num lock/caps lock keypresses themselves when in "Disambiguate escape codes" mode though, which kitty does not do.

The sequence "\<Num Lock on\>\<space\>\<Num Lock off\>\<space\>" with the procedure above is now giving
Kitty: \x20 \x20
Wezterm: \x1b[57360;1u \x20 \x1b[57360;129u \x20

**Comment by wez** [comment]
Should have been resolved by a562cb5a92697b180d0e50ca791248d73522aa1f, wherein I typo'd the issue number and it didn't automatically show up as referencing this issue.

**Comment by AThePeanut4** [comment]
Amazing, can confirm it now works as expected. Thanks!
